internal/vehicle/carwings.go

```diff
@@ -57,52 +77,180 @@ func (v *CarWings) initSession() error {
 	v.session = &carwings.Session{
 		Region: v.region,
 	}
-
+	v.refreshKey = ""
 	return v.session.Connect(v.user, v.password)
 }
 
-// chargeState implements the api.Vehicle interface
-func (v *CarWings) chargeState() (float64, error) {
+// carwingsVehicleStatus holds the relevant data extracted from carwings API
+// on vehicle status request
+type carwingsVehicleStatus struct {
+	VehicleStatus struct {
+		SOC            float64
+		Range          int64
+		ChargingStatus carwings.ChargingStatus
+		PlugStatus     carwings.PluginState
+		ClimateStatus  struct {
+			active      bool
+			targetTemp  float64
+			outsideTemp float64
+		}
+		LastRefresh time.Time
+	}
+}
+
+func (v *CarWings) status() (res carwingsVehicleStatus, err error) {
```

Comment:
why not return error only? the rest is handled by the carwings session.

Reply:
I changed it locally to return bool and error, because of the cache provider usage. How would this work for error only?

Reply:
You could wrap it or return an empty interface as first result

Reply:
OK, but then I would prefer the bool solution in the new commit. I tested it over the day and it was working fine,